src/track/beats.cpp

```diff
@@ -595,7 +597,12 @@ mixxx::Bpm Beats::getBpmAroundPosition(audio::FramePos position, int n) const {
 
     // To make sure we are always counting n beats, iterate backward to the
     // lower bound, then iterate forward from there to the upper bound.
-    it -= n;
+    for (int i = 0; i < n; ++i) {
+        if (it == cbegin()) {
+            break;
+        }
+        it--;
```

Comment:
I am unsure about this solution because it has performance implications. 
We go here step by step backward to not reach cbegin() and hit the assertion. 
The code is able to do it in one go. If we rely on this, we need to remove the warning and the assertion form the -= operator. 

What do you think?

Reply:
The same below, we first hit the assertion and than do the correction.